### PR TITLE
Support refresh token authentication

### LINKED ISSUE #55: Delegation for Firebase
Hi 🙂

I am using this library in an iOS App that needs authentication to Firebase with the id_token got by calling `FIRAuth.auth()?.signIn(withCustomToken:...)`

From what I understood I need to do the following steps in order to get this custom token:
- login to Auth0 with email and password to get the credentials via the `Auth0.authentication()        .login(...)` method
- with the credentials I get the profile via the `Auth0.authentication().tokenInfo(...) method`
- get the delegation token from the id_token in the profile

But I this library I cannot find any method to get a delegation token. Is this missing or I have misunderstood the logic ?

Thanks a lot 😉

### LINKED ISSUE #59: Missing API
The delegation endpoint API for refreshing the id_token is missing.

Is this planned? If so, when will this become available?

Cheers!

### PR DESCRIPTION
**Pull request by cocojoe**
Also adds method to call delegation.

Closes #55 also fixes #59

### CODE REVIEW

File: Auth0/Authentication.swift [hunk 1]
```diff
@@ -339,6 +371,14 @@ public protocol Authentication: Trackable, Loggable {
      - seeAlso: https://tools.ietf.org/html/rfc7636
      */
     func tokenExchange(withCode code: String, codeVerifier: String, redirectURI: String) -> Request<Credentials, AuthenticationError>
+
+    /**
+     Authenticate with a refresh_token.
```
Comment (hzalaz) [inline Auth0/Authentication.swift]: `Renew user's credentials with a refresh_token`

File: Auth0/Authentication.swift [hunk 2]
```diff
@@ -339,6 +371,14 @@ public protocol Authentication: Trackable, Loggable {
...
+
+     - Parameter token: the clients refresh_token
```
Comment (hzalaz) [inline Auth0/Authentication.swift]: `- Parameter refreshToken: the client's refresh token obtained on auth

File: Auth0/Authentication.swift [hunk 3]
```diff
@@ -339,6 +371,14 @@ public protocol Authentication: Trackable, Loggable {
...
+     - Returns: a request that will yield Auth0 user's credentials
+     */
+    func login(withToken token: String) -> Request<Credentials, AuthenticationError>
```
Comment (hzalaz) [inline Auth0/Authentication.swift]: `func renew(withRefreshToken refreshToken: String) -> Request<Credentials, AuthenticationError>`

Comment (hzalaz) [inline Auth0/Authentication.swift]: Needs to be `public`

Comment (cocojoe) [inline Auth0/Authentication.swift]: Can't change to public in protocol, access is inline with other methods and I didn't have any issues accessing in the App ViewController.

**Pull request merged**